### ISSUE TITLE
How does the `pos` option actually work?

### ISSUE THREAD

**Issue by SteffanDonal** [moudey/Shell#160]
Hey again!

I'm struggling to get menu items to show in a specific order. I'm working via "static" modification, and I've not investigated "dynamic" yet... Here's my problem.

Imagine this menu:
- MyFirstItem
- MySecondItem
- MyThirdItem
- MyFourthItem
- MyFifthItem

I want to move `MyThirdItem` at the bottom, and I want to move `MySecondItem` to be after `MyThirdItem`.
I'm expecting to see:
- MyFirstItem
- MyFourthItem
- MyFifthItem
- MyThirdItem
- MySecondItem

I can achieve this:
```
item(find="MyThirdItem" pos=3)
item(find="MySecondItem" pos=4)
```

However, **this will only work when there are 4 or 5 items in the menu.**

If there are more items, `MyThirdItem` will no longer be at the bottom.
_And if there are any fewer items, `MySecondItem` actually appears above `MyThirdItem`!_

The reason for this, I think, is simple, but the side-effect is that position adjustment is extremely limited. 

It seems that when setting any item's position to a positive integer that's "out of range", the item is instead assigned "bottom", and **all items with the same position value are sorted "as originally ordered".**

That last point is very hard to deal with; any and all custom positioning will be ignored when the numeric values provided are out of range for the current menu's length.

## What I'd like to be able to do

```
// Set all items to the same "base"
item(where=true pos=50)

// 25 group
item(find="MyFourthItem"  pos=25)
item(find="MyFirstItem" pos=25)

// 45 group
item(find="MySecondItem" pos=45)
item(find="MyThirdItem" pos=45)

// 10 group
item(find="MyFifthItem" pos=10)
```
Should produce:
- MyFifthItem
- MyFourthItem
- MyFirstItem
- MySecondItem
- MyThirdItem

If positioning worked this way, I could create groups that can dynamically grow/shrink contextually without manually calculating the correct indices and anticipate every possible item that might show in the menu.

I hope this makes sense and you see the usefulness!
  
  
PS. I've also noticed that the "special" values, which are negative, run from `-1` to `-5` (inclusive). `bottom=-1`, `top=-2`, `middle=-4`, `auto=-4`. When testing, it appears that `middle` should be `-3`, and I'm not sure what `-5` does, but it _does_ actually seem to work?

**Comment by moudey** [comment]
Hi @SteffanDonal

A position assignment of 0 or higher is the actual rank in the array range. If the number is higher than the number of items, it is placed at the end of the menu.

Negative values are represented as follows:
```
enum Position
{
  Auto    = -4,
  Middle  = -3,
  Top     = -2,
  Bottom  = -1
}
```
I will study your suggestion and try to address it

**Comment by SteffanDonal** [comment]
> If the number is higher than the number of items, it is placed at the end of the menu.

Yep! This is what I expected was the case - however if you set multiple items to top/bottom, the order of those items will be "original" - and there's no way to influence it currently. I'm guessing that's because internally, numbers that are out of range are assigned "-1".

It would be great if numbers higher than the number of items would be sorted according to their assigned number, not "original".
(This is how my example works)

**Comment by SteffanDonal** [comment]
Also, the last comment for now - positioning behaviour does not interact as expected between dynamic/static.

EG. A dynamic item with `pos=middle` will appear _below_ a static item with `pos=bottom`

**Comment by moudey** [comment]
I will review this topic and provide a copy for you to test

**Comment by moudey** [comment]
>A dynamic item with `pos=middle` will appear below a static item with `pos=bottom`

Use one of these methods to assign the value to the position `pos='top'` or `pos=pos.top`

**Comment by Austin001316** [comment]
Is this closed because it's being changed in next update? I'm also having some position related issues that just don't seem to be working right. I have 2 menus and one seems to be changing position depending on what I right click on even though I gave it a set position. 
`menu(mode="multiple" type="*" title=title.more_options image=icon.more_options pos=10) { }`
`menu(type='file|dir|drive|namespace|back' mode="multiple" title='File Management' image=\uE253 pos=3)`
This is my initial code for both
![image](https://user-images.githubusercontent.com/91763031/234937228-624be682-b707-4cad-8d71-45d5846bd175.png)
.Exe context menu
![image](https://user-images.githubusercontent.com/91763031/234937286-1c10a3a3-5f18-43e0-a744-5807ea5565a7.png)
Single folder context menu
![image](https://user-images.githubusercontent.com/91763031/234937350-e50ce232-a17c-459b-ba73-f87be210247f.png)
Multiple folder context menu

**Comment by moudey** [comment]
@Austin001316 Please refer to this #234
